cardano-client/src/Cardano/Client/Subscription.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Cardano.Client.Subscription (
+  subscribe
```

Comment:
Lets just do that in the next iteration, after one or more clients have been adapted to this
API it will be much clearer how it works.
I find it much easier to improve the API incrementally.

Reply:
ok